docs/modules/ROOT/pages/cop_api_v1_changelog.adoc

```diff
@@ -0,0 +1,258 @@
+= Cop API v0 to v1 API changes
+:doctype: book
+
+== Upgrade guide
+
+Your custom cops should continue to work in v1.
+
+Nevertheless it is suggested that you tweak them to use the v1 API by following the following steps:
+
+1) Your class should inherit from `RuboCop::Cop::Base` instead of `RuboCop::Cop::Cop`.
+
+2) Locate your calls to `add_offense` and make sure that you pass as the first argument either an `AST::Node`, a `::Parser::Source::Comment` or a `::Parser::Source::Range`, and no `location:` named parameter.
+
+[discrete]
+==== Example:
+
+[source,ruby]
+----
+# Before
+class MySillyCop < Cop
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node, location: :selector, message: "Be positive")
+    end
+  end
+end
+
+# After
+class MySillyCop < Base
+  def on_send(node)
+    if node.method_name == :-
+      add_offense(node.loc.selector, message: "Be positive")
```

Comment:
Note for another section, but what if we had a `node.selector`?

Reply:
Not too sure what you mean. If you pass `node`, it's a shortcut for `node.loc.expression` (the range of the whole node). In the example, `selector` is the `:-`. Check `ruby-parse -L -e "a - b"`

Reply:
Aha, my mistake, disregard then